packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -202,6 +206,8 @@ String? _artifactToFileName(Artifact artifact, Platform hostPlatform, [ BuildMod
       return 'font-subset$exe';
     case Artifact.constFinder:
       return 'const_finder.dart.snapshot';
+    case Artifact.flutterToolsFileGenerators:
+      return '';
```

Comment:
should we ever reach this line of code? If so, it seems unexpected to me that it would return an empty string. If not, can we throw an `UnimplementedError` instead?

Reply:
hmm, although it seems there is a precedence for returning an empty string?

Reply:
Yeah I'm just following what existing code does. When it's a path to a folder (not a file), the expected return value here is empty string.

Reply:
and also returning null apparently? I am still leaning towards throwing here, unless there is some legitimate use case for `''`

Reply:
I did it in a separate PR since I don't want to delay this PR any further. Here: https://github.com/flutter/flutter/pull/120484

Reply:
Oh I didn't catch that you suggested throwing `UnimplementedError`. Do you prefer that or an assertion?